### PR TITLE
GCW:2397 Migrate shared.goodcity to CI 2.0

### PR DESCRIPTION
**Pull request by Darshan-Chauhan**
Hi Team,

This PR migrates shared.goodcity to circleci 2.0
Jiraticket:- https://jira.crossroads.org.hk/browse/GCW-2397

Please review.

### CODE REVIEW

**Comment by Darshan-Chauhan** [comment]
Hi All, 
I'm merging this to check if new CircleCI api triggers Admin and App master/live branch workflow.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      paths:
+        - bower_components
+
+jobs:
+  checkout_code:
+    <<: *defaults
+    steps:
+      - checkout
+      - save_cache:
+          key: shared-goodcity-repo-{{ .Environment.CIRCLE_SHA1 }}
```
Comment (steveyken) [inline .circleci/config.yml]: @Darshan-Chauhan  Is it worth moving this key up into the aliases section next to where it is already defined? (Small detail)

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            - ~/code
+
+  package_dependencies:
+    <<: *defaults
+    steps:
+      - restore-cache: *restore-repo-cache
+      - restore-cache: *restore-yarn-cache
+      - run: yarn link
+      - run: yarn add codeclimate-test-reporter
+      - run: yarn add bower
```
Comment (steveyken) [inline .circleci/config.yml]: Would it work to add codeclimate-test-report and bower to the package.json/yarn lockfile so they are automatically included in the build? This should save us some precious build time due to bundle caching.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+          name: Start Ember test server
+          command: yarn run ember server --port 4203
+          background: true
+      - run:
+          name: Wait for Ember test server to start
+          command: dockerize -wait tcp://localhost:4203 -timeout 1m
+      - run:
+          name: Run Ember tests
+          command: COVERAGE=true yarn run ember test
+      - persist_to_workspace:
```
Comment (steveyken) [inline .circleci/config.yml]: I think we are missing a step here to push the coverage report to Code Climate?

**Comment by Darshan-Chauhan** [comment]
@steveyken , thanks for reviewing. I'll add the changes.

**Pull request merged**